sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/StateStoreSuite.scala

```diff
@@ -1373,6 +1386,12 @@ abstract class StateStoreSuiteBase[ProviderClass <: StateStoreProvider]
           eventually(timeout(30.seconds)) {
             assert(!StateStore.isMaintenanceRunning)
           }
+
```

Comment:
For a future-proof, might be better to reset the value of `FakeStateStoreProviderWithMaintenanceError.errorOnMaintenance` at the earlier of testing.

Reply:
Done